sql/core/src/main/java/org/apache/spark/sql/execution/datasources/parquet/VectorizedRleValuesReader.java

```diff
@@ -211,24 +212,25 @@ public void readIntegersWithRebase(
       WritableColumnVector c,
       int rowId,
       int level,
-      VectorizedValuesReader data) throws IOException {
+      VectorizedValuesReader data,
+      final boolean failIfRebase) throws IOException {
```

Comment:
Just in case, what is purpose of `final` here?

Reply:
means it's immutable and may help to optimize `if (failIfRebase)`